openapi/openapi_spec.json

```diff
@@ -15467,6 +15499,49 @@
           "disabled"
         ]
       },
+      "RecurringDetails": {
+        "oneOf": [
+          {
+            "type": "object",
+            "required": [
+              "type",
+              "data"
+            ],
+            "properties": {
+              "type": {
+                "type": "string",
+                "enum": [
+                  "mandate_id"
+                ]
+              },
+              "data": {
+                "type": "string"
+              }
+            }
+          },
+          {
+            "type": "object",
+            "required": [
+              "type",
+              "data"
+            ],
+            "properties": {
+              "type": {
+                "type": "string",
+                "enum": [
+                  "payment_method_id"
+                ]
+              },
+              "data": {
+                "type": "string"
+              }
+            }
+          }
+        ],
+        "discriminator": {
+          "propertyName": "type"
+        }
+      },
```

Comment:
```suggestion
      "RecurringDetails": {
        "type": "object",
        "required": [
          "type",
          "data"
        ],
        "properties": {
          "type": {
            "type": "string",
            "enum": [
              "mandate_id",
              "payment_method_id"
            ]
          },
          "data": {
            "type": "string"
          }
        },
        "discriminator": {
          "propertyName": "type"
        }
      }
```

Reply:
currently we are providing support to pass `mandate_id` inside `recurring_details`. So it has to be one of `mandate_id` or `payment_method_id`. Later we'll deprecate the `mandate_id` field in root level. 

> I hope the testing for recurring payment with mandate_id in root level and payment_token is done already
yes